src/pages/Governance/ProposalList/CreateProposalModal/index.tsx

```diff
@@ -40,86 +44,162 @@ export const CreateProposal: React.FC<CreateProposalProps> = ({
+            displayableErrorCodes={[ErrorCode.VALUE_REQUIRED]}
+          />
+
+          <FormikTextField
+            name="againstDescription"
+            placeholder={t('vote.createProposalForm.againstOption')}
+            label={t('vote.against')}
+            css={styles.formBottomMargin}
+            displayableErrorCodes={[ErrorCode.VALUE_REQUIRED]}
+          />
+
+          <FormikTextField
+            name="abstainDescription"
+            placeholder={t('vote.createProposalForm.abstainOption')}
+            label={t('vote.abstain')}
+            css={styles.sectionSpacing}
+            displayableErrorCodes={[ErrorCode.VALUE_REQUIRED]}
+          />
+        </>
+      ),
+    },
+    {
+      title: t('vote.pages.actions'),
+      Component: () => <ActionAccordion />,
+    },
+    {
+      title: t('vote.pages.confirmation'),
+      Component: () => <ProposalPreview />,
+    },
+  ];
+
+  const handleImportProposal = async (file: File | null, formikContext: ProposalFormContext) => {
+    const { setFieldValue, validateForm } = formikContext;
+    if (file) {
+      try {
+        const {
+          actions,
+          proposalType,
+          title,
+          description,
+          forDescription,
+          abstainDescription,
+          againstDescription,
+        } = await importJsonProposal(file);
+
+        setFieldValue('actions', actions);
+        setFieldValue('title', title);
+        setFieldValue('proposalType', proposalType);
+        setFieldValue('description', description);
+        setFieldValue('forDescription', forDescription);
+        setFieldValue('abstainDescription', abstainDescription);
+        setFieldValue('againstDescription', againstDescription);
+        await validateForm();
+        setProposalMode('file');
+        setCurrentStep(1);
+      } catch (error) {
+        let { message } = error as Error;
+
+        if (error instanceof VError) {
+          message = formatVErrorToReadableString(error);
+        }
+
+        toast.error({ message });
+      }
+    }
+  };
+
+  const steps = useMemo(() => {
+    const initialSteps = [
       {
-        title: t('vote.pages.confirmation'),
-        Component: () => <ProposalPreview />,
+        title: t('vote.createProposalModal.createProposal'),
+        Component: () => {
+          const formikContext = useFormikContext<FormValues>();
+          return (
+            <UploadOrManualProposal
+              onClickCreateManually={() => {
+                setProposalMode('manual');
+                setCurrentStep(1);
+              }}
+              onClickUploadFile={async file => handleImportProposal(file, formikContext)}
+            />
+          );
+        },
       },
-    ],
-    [],
-  );
+    ];
+    if (proposalMode === 'file') {
+      return initialSteps.concat(uploadFileSteps);
+    }
+    if (proposalMode === 'manual') {
+      return initialSteps.concat(manualProposalSteps);
+    }
+    return initialSteps;
+  }, [proposalMode]);
```

Comment:
Seeing the logic get more complicated, how about using nested routing to handle the navigation within the Create Proposal modal? https://v5.reactrouter.com/web/example/nesting